public/index.html

```diff
@@ -1,121 +1,247 @@
+          </svg>
+        </div>
+        App Engine
+      </div>
+      <div class="option nota">
+        <div class="checkbox">
+          <svg xml:space="preserve" focusable="false" version="1.1" viewBox="0 0 24 24">
+            <path class="checkmark-path" d="M4.1,12.7 9,17.6 20.3,6.3" fill="none"></path>
+          </svg>
+        </div>
+        None of the above
+      </div>
+      <div class="submit-btn">Submit</div>
+    </div>
+
+    <h2 class="thanks">Thanks for voting! <br>Checkout the <a href="/results">results</a>.</h2>
+
+  </div>
+
 
-    document.querySelector('#submit').classList.add('loading');
-
-    fetch(url, {
-      method: 'POST',
-      body: JSON.stringify(votes),
-      headers: {
-        'Content-Type': 'application/json; charset=utf-8'
-      },
-    }).then((response) => {
-      document.querySelector('#submit').classList.remove('loading');
-      console.log(response);
-      if (response.ok) {
-        document.querySelector('#options').style.display = 'none';
-        document.querySelector('#middle').style.display = 'table';
+  <script>
+    const optionData = ['Skaffold', 'Jib', 'Knative', 'App Engine', 'None of the above'];
+    const survey = document.querySelectorAll('.survey');
+    const thanks = document.querySelector('.thanks');
+    const options = document.querySelectorAll('.option');
+    const submitBtn = document.querySelector('.submit-btn');
+
+    options.forEach((el) => el.addEventListener('click', clickHandler));
+
+    function clickHandler(event) {
+      let el = event.target;
+      if (el.classList.contains('selected')) {
+        el.classList.remove('selected');
       } else {
-        alert('Error: Request failed with Status Code: ' + response.status);
+        if(el.classList.contains('nota')) {
+          options.forEach(el => el.classList.remove('selected'));
+        }
+        else {
+          document.querySelector('.nota').classList.remove('selected');
+        }
+        el.classList.add('selected');
       }
+    }
+
+    submitBtn.addEventListener('click', () => {
+      if (submitBtn.classList.contains('loading')) return;
+      submitBtn.classList.add('loading');
+
+      const url = '/vote';
+      const votes = {};
+
+      options.forEach((el) => {
+        if (el.classList.contains('selected')) {
+          votes[el.innerText] = 1;
+        }
+      });
+
+      console.log(votes);
+
+      fetch(url, {
+        method: 'POST',
+        body: JSON.stringify(votes),
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8'
+        },
+      }).then((response) => {
+        submitBtn.classList.remove('loading');
+        console.log(response);
+        if (response.ok) {
+          survey.style.display = 'none';
+          thanks.style.display = 'block';
+        } else {
+          console.log('Error: Request failed with Status Code: ' + response.status);
```

Comment:
Can we show this error to the user in the UI?
As part of the demo, we explicitly to fail this request and fix it. So we want users to see the failure.